int32 hue_max
int32 hue_min
int32 saturation_max
int32 saturation_min
int32 value_max
int32 value_min                                                 